File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/persistence/cdmfolder/types/projections/Projection.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.types.projections;

import java.util.List;

/**
 * Projection class
 */
public class Projection {
    private String explanation;
    private String condition;
    private List<OperationBase> operations;
    private Object source;
    private Boolean runSequentially;

    public String getExplanation() {
        return explanation;
    }

    public void setExplanation(final String explanation) {
        this.explanation = explanation;
    }

    public String getCondition() {
        return condition;
    }

    public void setCondition(final String condition) {
        this.condition = condition;
    }

    public List<OperationBase> getOperations() {
        return operations;
    }

    public void setOperations(final List<OperationBase> operations) {
        this.operations = operations;
    }

    public Object getSource() {
        return source;
    }

    public void setSource(final Object source) {
        this.source = source;
    }

    public Boolean getRunSequentially() {
        return runSequentially;
    }

    public void setRunSequentially(Boolean runSequentially) {
        this.runSequentially = runSequentially;
    }
}

File: objectModel/TypeScript/Persistence/CdmFolder/types/Projections/Projection.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import { EntityReferenceDefinition } from '../EntityReferenceDefinition';
import { OperationBase } from './OperationBase';

/**
 * Projection class
 */
export abstract class Projection {
    public explanation: string;
    public condition: string;
    public operations: OperationBase[];
    public source: EntityReferenceDefinition;
    public runSequentially?: boolean;
}
